Src/AutoFixture/EmailAddressLocalPart.cs

```diff
@@ -0,0 +1,142 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Ploeh.AutoFixture
+{
+    /// <summary>
+    /// Represents the local part of the email address, defined as everything up to, but not including the @ sign.  
+    /// The requirements enforced on the local part are defined in RFC 3696.  The regular expression for validating 
+    /// the local part borrowed from Phil Haack at 
+    /// http://haacked.com/archive/2007/08/21/i-knew-how-to-validate-an-email-address-until-i.aspx/
```

Comment:
It depends on how we view the utility of `EmailAddressLocalPart` within AutoFixture.  If we were writing a library for validating email addresses, it may make sense to do this level of validation.  However, as we've seen, even deciding to do that isn't exactly simple since there can definitely exist email addresses in use that aren't valid per the RFC so seems there's no actual standard that could be applied that wouldn't be overly restrictive.  

Within AutoFixture, I would think we'd use the local part more loosely to essentially mean "some token on the left side of the @ sign" or else we could make it very difficult to actually use `EmailAddressLocalPart`.  So I propose we have some basic rules, but otherwise allow a great deal of latitude in its content.  I'm thinking something along the lines of length <= 64, cannot contain @ sign, cannot contain whitespace characters, but otherwise any other character is fine.  I don't think I would even enforce the rules around periods.  These proposed rules are arbitrary, somewhat based on what I perceive as the core of the local part spec, but very liberal in terms of the character content.  We could also have no rules whatsoever other than not allowing it to be null or blank.  I'm open to however you'd like this to go.